examples/motoko_examples/whoami/src/whoami.ts

```diff
@@ -0,0 +1,75 @@
+import {
+    Canister,
+    CanisterResult,
+    ic,
+    Init,
+    PostUpgrade,
+    Principal,
+    Query,
+    UpdateAsync,
+    Variant
+} from 'azle';
+
+type WhoAmICanister = Canister<{
+    installer(): CanisterResult<Principal>;
+    argument(): CanisterResult<Principal>;
+    whoami(): CanisterResult<Principal>;
+    id(): CanisterResult<Principal>;
+    idQuick(): CanisterResult<Principal>;
+}>;
+
+type WhoAmIResult = Variant<{
+    ok: Principal;
+    err: string;
+}>;
+
+// Initialize the variables to ensure that they aren't `undefined`.
+// We use the zero address but any principal could be used.
```

Comment:
This comment may not technically correct, it's the zero principal not the zero address. Addresses are different concepts and this could lead to confusion.

Reply:
Updated `address` to `principal`. We should be good now.